pkg/apply/prune/inventorycm.go

```diff
@@ -0,0 +1,135 @@
+// Copyright 2020 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Introduces the InventoryConfigMap struct which implements
+// the Inventory interface.
+
+package prune
+
+import (
+	"fmt"
+	"sort"
+	"strconv"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/cli-runtime/pkg/resource"
+	"sigs.k8s.io/cli-utils/pkg/common"
+	"sigs.k8s.io/cli-utils/pkg/object"
+)
+
+func WrapInventoryObj(info *resource.Info) Inventory {
+	return &InventoryConfigMap{inv: info}
+}
+
+type InventoryConfigMap struct {
+	inv      *resource.Info
+	objMetas []*object.ObjMetadata
+}
+
+func (icm *InventoryConfigMap) Load() ([]*object.ObjMetadata, error) {
+	objs := []*object.ObjMetadata{}
```

Comment:
I don't think we should use pointers here.

Reply:
Fixed